packages/browser/src/authenticatedFetch/fetchFactory.ts

```diff
@@ -42,6 +42,26 @@ export function buildBearerFetch(authToken: string): typeof fetch {
   };
 }
 
+async function buildDpopFetchOptions(
+  targetUrl: string,
+  authToken: string,
+  dpopKey: JSONWebKey,
+  defaultOptions?: RequestInit
+): Promise<RequestInit> {
+  return {
+    ...defaultOptions,
```

Comment:
So how does this work? I'd have expected there to be a `redirect: 'manual'` in here somehow, but this looks like a redirect returns a successful response with a different `.url` property?

Reply:
I initially included `redirect: "manual"`, but that actually did not make a difference in the implementation, because the `redirected` property of the Response wasn't set how I expected it to be. So it's as you say: `fetch` follows the redirects "normally", and eventually returns a Response. You don't have access to the intermediate redirected responses, so from the client's point of view you'll never see a 301 status code for instance, but the final response does have an `.url` property that makes it possible to detect the actual IRI of the target resource. From there, we can just re-issue the request.

Reply:
Hmm OK, I suppose technically that would work, but I'm still not 100% comfortable with it. I suppose this does mean that the target of the redirect will receive two requests rather than one, with the first one failing?

I did some tests locally, and although it seems like `redirected` is always `false` if you set `redirect` to `"manual"`, but the `type` of the Response does indicate whether you need to do a redirect or not, i.e. it will be `"opaqueredirect"` if you do.

To reproduce the tests, I created an empty dir, ran `npm install express`, then created the following `index.js`:

```js
const express = require('express')
const app = express()
const port = 3000

app.get('/', (req, res) => {
  res.sendStatus(200);
})

app.get('/redirectfail', (req, res) => {
  res.redirect(301, "http://localhost:3000/fail")
})

app.get('/fail', (req, res) => {
  res.sendStatus(403);
})

app.get('/redirectsuccess', (req, res) => {
  res.redirect(301, "http://localhost:3000/success")
})

app.get('/success', (req, res) => {
  res.send("Hoi");
})

app.listen(port, () => {
  console.log(`Example app listening at http://localhost:${port}`)
})
```

You can then see what happens in the different scenarios using your browser console:

1a:
```
await fetch("http://localhost:3000/success", { redirect: "manual" });
// redirected: false
// type: "basic"
// url: "http://localhost:3000/success"
```


1b:
```
await fetch("http://localhost:3000/redirectsuccess", { redirect: "manual" });
// redirected: false
// type: "opaqueredirect"
// url: "http://localhost:3000/redirectsuccess"
```

1c:
```
await fetch("http://localhost:3000/redirectfail", { redirect: "manual" });
// redirected: false
// type: "opaqueredirect"
// url: "http://localhost:3000/redirectfail"
```

2a:
```
await fetch("http://localhost:3000/success");
// redirected: false
// type: "basic"
// url: "http://localhost:3000/success"
```

2b:
```
await fetch("http://localhost:3000/redirectsuccess");
// redirected: true
// type: "basic"
// url: "http://localhost:3000/success"
```

2c:
```
await fetch("http://localhost:3000/redirectfail");
// redirected: true
// type: "basic"
// url: "http://localhost:3000/fail"
```

Essentially, what we do is distinguish b and c (i.e. redirects) from a. Without `redirect: "manual" (i.e. the scenarios labeled 2)`, indeed the fact that the `url` on the Response differs from the requested URL distinguishes them.

However, when the target is not accessible (i.e. 2c) the first redirect call to `fetch` already logs an error, and then the DPoP retry will log another one. This does not happen when `redirect: "manual"` is set (1c).

Then we'd need to distinguish 1b and 1c (redirects) from 1a (a regular request). That, apparently, can be done by checking for `response.type === "opaqueredirect"`.

Reply:
Thanks for the explanation, that makes it very clear ! 

I have a concern though: in the case where we don't set `redirect: manual`, the browser handles the redirections. In the case we handle them manually, we have to manage the case where multiple redirections happen in a row (`http://movedOnce.net/resource` -> `http://movedTwice.net/resource` -> `http://target.com/resource`). However, what happens if there is a cycle ? Should we have a maximum number of jumps ? If you add 
```
app.get('/redirectloop', (req, res) => {
  res.redirect(301, "http://localhost:3000/redirectloop")
})
```
to the previous server, it looks like without `redirect: manual`, the browser throws an error after 21 requests, so 20 redirections. If that's good enough for Firefox, that's good enough for me :p

Reply:
Proposed solution: 85dc450819d9cad8fa2959016a2b57061de0a4ec

Reply:
Wait, I just realised there was is issue: when using `redirect: manual`, you don't have the IRI to which you should be redirected, `response.url` isn't updated to show the new target, so there's no way to know what the request should be redirected to. I'll change the code back to how it was before, but I'll see if I can be smartest with the server's response in order not to replay a request that is failing for non-auth related issues.

Reply:
https://github.com/whatwg/fetch/issues/763 🤦‍♂️

Reply:
Fixed in 5a9b9cd22fce6f79e5e5a9cb59ed272e6df7bd2f

Reply:
Dammit. OK, well, if the e2e test succeeds, that should give enough confidence then, I suppose. Sorry for the extra work!